### PR TITLE
Modified the shell utility

### PR DESCRIPTION
**Pull request by etn406**
### Small changes
- Args are now handled by [clap](https://docs.rs/clap/latest/clap/)
- There is now a nice `--help` thanks to _clap_.
- `--hostport` can be shortened to `-h` and `--password` to `-p`.
- Minecraft format codes (ie. `$9`) are now removed by default from the output (I also added an flag `--keep-formatting` or `-k` to avoid that removal).

### Bigger changes
The utility now has two behaviours :
- By default, it send a single command, display the response and exits : 
```
$ ./cargo run -- "time set day"
L'heure a été réglée à 0 ticks dans : world, world_nether, world_the_end
$
``` 
- Or the interactive shell is still accessible with the flag `--interactive` or `-i`
```
$ ./cargo run -- -i
Starting RCON shell. Use 'exit', 'quit', or Ctrl-C to exit.
>
``` 

------

I'm a beginner with Rust so do not hesitate to point me the issues in this PR 😄

### CODE REVIEW

File: src/bin/shell.rs [hunk 1]
```diff
@@ -1,104 +1,163 @@
-use std::env;
+use clap::{ArgGroup, Parser};
```
Comment (willroberts) [inline src/bin/shell.rs]: Great addition, thanks!

File: Cargo.toml [hunk 2]
```diff
@@ -7,3 +7,5 @@ description = "A client for the Minecraft RCON protocol."
 license = "GPL-3.0-or-later"
 
 [dependencies]
+clap = { version = "3.2", features = ["derive"] }
+regex = "1.6"
```
Comment (willroberts) [inline Cargo.toml]: Can we gate these two new dependencies behind a `cli` feature? https://doc.rust-lang.org/cargo/reference/features.html

**Review by willroberts — CHANGES_REQUESTED** [review]

**Comment by willroberts** [comment]
Perfect, thanks again!

**Review by willroberts — APPROVED** [review]

**Comment by etn406** [comment]
You're welcome, I'm happy I did that!

Do you take care of the update of `README.md`concerning the use of the shell utility, or do I do it?

**Comment by willroberts** [comment]
> Do you take care of the update of `README.md`concerning the use of the shell utility, or do I do it?

I'll take care of this today. I will also publish v0.1.3 (with README updates) and publish to crates.io.

**Comment by willroberts** [comment]
Updated!

New release: https://github.com/willroberts/minecraft-client-rs/releases/tag/v0.1.3
New crate: https://crates.io/crates/minecraft-client-rs/0.1.3

**Pull request merged**